### PR TITLE
CASMPET-5117: Update setup scripts to better support admin auth

### PR DESCRIPTION
**Pull request by zcrisler**
Updates Nexus setup scripts to more securely support custom admin credentials. Also adds some corresponding `nexus-disable` scripts. 

Updates hack scripts to mirror functionality in the `cray-nexus` Helm chart:
- `nexus-init` mirrors the functionality in the post-install hook that just cleans up the default state and sets the local admin password
- `hack/nexus-setup` and `hack/nexus-teardown` mirror the functionality in the Helm post-install/upgrade hook to enable/disable specific functionality

Tested locally using the `hack` scripts as follows:

```
$ ./hack/nexus-create
$ ./hack/nexus-init
$ ./hack/nexus-setup
...
$ ./hack/nexus-teardown
...
$ ./hack/nexus-delete
```

### CODE REVIEW

File: bin/nexus-disable-keycloak-realm [hunk 1]
```diff
@@ -0,0 +1,19 @@
+#!/usr/bin/env bash
+
+# Copyright 2022 Hewlett Packard Enterprise Development LP
+
+command -v jq >/dev/null 2>&1 || { echo >&2 "command not found: jq"; exit 1; }
+
+. "$(dirname "$0")/nexus-ready"
+
+echo -n >&2 "Disable Keycloak Authenticating security realm..."
+
```
Comment (rnoska-hpe) [inline bin/nexus-disable-keycloak-realm:10]: Does this need to be idempotent?  What is the effect in the case where the item is missing?

File: bin/nexus-enable-rut-auth [hunk 2]
```diff
@@ -1,12 +1,12 @@
...
 
-# Copyright 2020 Hewlett Packard Enterprise Development LP
+# Copyright 2022 Hewlett Packard Enterprise Development LP
 
 . "$(dirname "$0")/nexus-ready"
 
 # requires the setup_capability script
 
-echo -n >&2 "Enable Rut Auth..."
+echo -n >&2 "Enable Rut Auth capability..."
```
Comment (rnoska-hpe) [inline bin/nexus-enable-rut-auth:9]: Is this idempotent?  What is the effect of running this multiple times?

File: bin/nexus-disable-keycloak-realm (hunk 1 above)
Comment (zcrisler) [reply bin/nexus-disable-keycloak-realm:10]: If the Keycloak realm isn't active, this action is effectively a no-op. It downloads the set of active realms, there is nothing to delete, and then puts back the same set of active realms. You can verify this behavior by running `hack/nexus-teardown` repeatedly after having run `hack/nexus-create` and `hack/nexus-init`.

File: bin/nexus-enable-rut-auth (hunk 2 above)
Comment (zcrisler) [reply bin/nexus-enable-rut-auth:9]: Running multiple times simply reconfigures the capability (i.e., enables it and sets the HTTP header to `X-WEBAUTH-USER`). You can verify the behavior by running `hack/nexus-setup` repeatedly.

Comment (zcrisler) [reply bin/nexus-enable-rut-auth:9]: I just updated this script to make the `httpHeader` customizable.

**Comment by zcrisler** [comment]
Successfully tested a pre-release tag of the updated image as part of a Nexus upgrade on wasp.

**Review by rnoska-hpe — APPROVED** [review]
Approval based on testing done on Wasp.

**Pull request merged**